spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyRestSupportAutoConfiguration.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.spanner;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.rest.webmvc.spi.BackendIdConverter;
+
+/**
+ * @author Balint Pato
+ */
+
+@Configuration
+@AutoConfigureAfter(GcpSpannerAutoConfiguration.class)
+@ConditionalOnMissingBean({ BackendIdConverter.class })
+@ConditionalOnClass({ BackendIdConverter.class })
+public class SpannerKeyRestSupportAutoConfiguration {
+
+	@Bean
+	@ConditionalOnMissingBean
+	public BackendIdConverter backendIdConverter() {
```

Comment:
Method names are picked as bean names. It can be unfortunate to pick a generic name and the generic `BackendIdConverter`. Anyone registering a `BackendIdConverter` will stop this bean definition from contributing the bean. Exposing `SpannerKeyIdConverter spannerKeyIdConverter()` would be a safer approach.